web/src/components/__tests__/ContrastThemeToggle.spec.tsx

```diff
@@ -0,0 +1,74 @@
+import { fireEvent } from '@testing-library/react'
+import { mocked } from 'jest-mock'
+import React from 'react'
+
+import { useThemeContext } from '../../hooks/useThemeContext'
+import useWindowDimensions from '../../hooks/useWindowDimensions'
+import { renderWithTheme } from '../../testing/render'
+import { mockWindowDimensions } from '../../testing/utils'
+import ContrastThemeToggle from '../ContrastThemeToggle'
+import { ThemeContainer } from '../ThemeContext'
+
+jest.mock('../../hooks/useWindowDimensions', () => jest.fn(() => ({ viewportSmall: false })))
+jest.mock('react-i18next')
+
+const TestThemeTypeChange = () => {
+  const { themeType } = useThemeContext()
+  return <div data-testid='theme-type'>{themeType}</div>
+}
+
+const toggleContrastTheme = () =>
+  renderWithTheme(
+    <ThemeContainer contentDirection='ltr'>
+      <>
+        <ContrastThemeToggle />
+        <TestThemeTypeChange />
+      </>
+    </ThemeContainer>,
+  )
+
+describe('ContrastThemeToggle', () => {
+  beforeEach(jest.clearAllMocks)
+
+  it('render toggle button in mobile view', () => {
+    mocked(useWindowDimensions).mockImplementation(() => ({ ...mockWindowDimensions, viewportSmall: true }))
+    const { getByRole } = toggleContrastTheme()
+    const toggleButton = getByRole('button', { name: 'layout:contrastTheme' })
+    expect(toggleButton).toBeInTheDocument()
+  })
+
+  it('toggles theme between light and contrast in mobile view', () => {
+    mocked(useWindowDimensions).mockImplementation(() => ({ ...mockWindowDimensions, viewportSmall: true }))
+    const { getByRole, getByTestId } = toggleContrastTheme()
+    const toggleButton = getByRole('button', { name: 'layout:contrastTheme' })
+    const themeType = getByTestId('theme-type')
+    expect(themeType).toHaveTextContent('light')
```

Comment:
🤔 Any specific reason why you need the test id instead of just querying directly with `getByText('light')?

Reply:
no specific reason. You are right...I could just use your option :+1: